apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetItemDAO.java

```diff
@@ -91,69 +83,26 @@ INSERT INTO dataset_items (
-                    multiIf(
-                        LENGTH(CAST(old.dataset_id AS Nullable(String))) > 0 AND notEquals(old.dataset_id, new.dataset_id), leftPad('', 40, '*'),
-                        LENGTH(CAST(old.dataset_id AS Nullable(String))) > 0, old.dataset_id,
-                        new.dataset_id
-                    ) AS dataset_id,
-                    new.source,
-                    new.trace_id,
-                    new.span_id,
-                    new.input,
-                    new.expected_output,
-                    new.metadata,
-                    multiIf(
-                        notEquals(old.created_at, toDateTime64('1970-01-01 00:00:00.000', 9)), old.created_at,
-                        new.created_at
-                    ) AS created_at,
-                    multiIf(
-                        LENGTH(old.workspace_id) > 0 AND notEquals(old.workspace_id, new.workspace_id), CAST(leftPad('', 40, '*') AS FixedString(19)),
-                        LENGTH(old.workspace_id) > 0, old.workspace_id,
-                        new.workspace_id
-                    ) AS workspace_id,
-                    if(
-                        LENGTH(old.created_by) > 0, old.created_by,
-                        new.created_by
-                    ) AS created_by,
-                    new.last_updated_by
-                FROM (
+                VALUES
                     <items:{item |
-                         SELECT
-                             :id<item.index> AS id,
-                             :datasetId<item.index> AS dataset_id,
-                             :source<item.index> AS source,
-                             :traceId<item.index> AS trace_id,
-                             :spanId<item.index> AS span_id,
-                             :input<item.index> AS input,
-                             :expectedOutput<item.index> AS expected_output,
-                             :metadata<item.index> AS metadata,
-                             now64(9) AS created_at,
-                             :workspace_id AS workspace_id,
-                             :createdBy<item.index> AS created_by,
-                             :lastUpdatedBy<item.index> AS last_updated_by
+                        (
+                             :id<item.index>,
+                             :datasetId<item.index>,
+                             :source<item.index>,
+                             :traceId<item.index>,
+                             :spanId<item.index>,
+                             :input<item.index>,
+                             :expectedOutput<item.index>,
+                             :metadata<item.index>,
+                             now64(9),
+                             :workspace_id,
+                             :createdBy<item.index>,
+                             :lastUpdatedBy<item.index>
+                         )
                          <if(item.hasNext)>
-                            UNION ALL
+                            ,
```

Comment:
Much nicer like this since multiple rows are allowed in a single statement: 
- https://clickhouse.com/docs/en/sql-reference/statements/insert-into
Well done!